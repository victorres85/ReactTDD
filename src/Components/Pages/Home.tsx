import Form from "../Form";
import ParticipantsList from "../PaticipantsList";
import Footer from "../Footer";

const Home = () => {
  return (
    <>
      <Form />
      <ParticipantsList />
      <Footer />
    </>
  );
};

export default Home;
