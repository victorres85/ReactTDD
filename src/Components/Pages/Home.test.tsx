import { RecoilRoot } from "recoil";
import Home from "./Home";
import { render } from "@testing-library/react";

const mockNavegation = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavegation,
  };
});

describe("Home Page", () => {
  test("Home Page has to render all the relevant elements", () => {
    const { container } = render(
      <RecoilRoot>
        <Home />
      </RecoilRoot>,
    );
    expect(container).toMatchSnapshot();
  });
});
