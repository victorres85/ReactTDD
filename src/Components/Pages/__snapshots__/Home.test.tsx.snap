// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Home Page has to render all the relevant elements 1`] = `
<div>
  <form>
    <input
      placeholder="Add Paticipants' names"
      type="text"
      value=""
    />
    <button
      disabled=""
    >
      Add
    </button>
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Draw Names
    </button>
  </footer>
</div>
`;
